cmd/karmada-search/app/options/options.go

```diff
@@ -87,6 +87,9 @@ func (o *Options) Run(ctx context.Context) error {
 
 	server.GenericAPIServer.AddPostStartHookOrDie("start-karmada-search-informers", func(context genericapiserver.PostStartHookContext) error {
 		config.GenericConfig.SharedInformerFactory.Start(context.StopCh)
+
+		// start ResourceRegistry controller
+		search.NewController(restConfig, search.DefaultResourceHandler()).Start(context.StopCh)
```

Comment:
@RainbowMango Start from here, what about it?

Reply:
cc @XiShanYongYe-Chang to confirm.

Reply:
Starting ResourceRegistry controller from here is ok!

We can also add another PostStartHookFunc to `GenericAPIServer` with its own name just like:

```go
server.GenericAPIServer.AddPostStartHookOrDie("start-resourceRegistry-controller", func(context genericapiserver.PostStartHookContext) error {
		ctl, err := search.NewController(restConfig, search.CachedResourceHandler())
		if err != nil {
			return err
		}
		ctl.Start(context.StopCh)
		return nil
	})
```